test/network/scenarios/common.js

```diff
@@ -78,13 +81,53 @@ module.exports = {
 	getAllPeers,
 
 	stopNode: nodeName => {
-		return childProcess.execSync(`pm2 stop ${nodeName}`);
+		return new Promise((resolve, reject) => {
+			childProcess.exec(`pm2 stop ${nodeName}`, err => {
+				if (err) {
+					return reject(
+						new Error(`Failed to stop node ${nodeName}: ${err.message}`)
+					);
+				}
+				resolve();
+			});
+		});
 	},
 	startNode: nodeName => {
-		childProcess.execSync(`pm2 start ${nodeName}`);
+		return new Promise((resolve, reject) => {
+			childProcess.exec(`pm2 start ${nodeName}`, err => {
+				if (err) {
+					return reject(
+						new Error(`Failed to start node ${nodeName}: ${err.message}`)
+					);
+				}
+				const nodeReadyTimeout = setTimeout(() => {
+					pm2LogProcess.stdout.removeAllListeners('data');
+					pm2LogProcess.removeAllListeners('error');
+					reject(new Error(`Failed to start node ${nodeName} before timeout`));
+				}, NODE_READY_TIMEOUT);
+
+				let pm2LogProcess = childProcess.spawn('pm2', ['logs', nodeName]);
+				pm2LogProcess.once('error', err => {
+					clearTimeout(nodeReadyTimeout);
+					pm2LogProcess.stdout.removeAllListeners('data');
```

Comment:
why not remove error listener here as well?

Reply:
It uses the `once('error', ...)` handler so if the handler executes once then it will already have removed itself implicitly. I can change it to `on('error', ...)` and then explicitly remove the 'error' event handler if you prefer.

Reply:
Also note that I moved up the `let pm2LogProcess = ...` declaration since that's more clear. Also I made it `const`.